File: android/src/main/java/club/medalz/plugins/intagramshareplugin/InstagramSharePlugin.java
package club.medalz.plugins.intagramshareplugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "InstagramShare")
public class InstagramSharePlugin extends Plugin {

    private InstagramShare implementation = new InstagramShare();

    @PluginMethod
    public void shareToInstagramStories(PluginCall call) {
        implementation.shareToInstagramStories(call);
    }
}

File: ios/Sources/InstagramSharePlugin/InstagramSharePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InstagramSharePlugin)
public class InstagramSharePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "InstagramSharePlugin"
    public let jsName = "InstagramShare"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "shareToInstagramStories", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = InstagramShare()

    @objc func shareToInstagramStories(_ call: CAPPluginCall) {
        implementation.shareToInstagramStories(call)
    }
}
